src/Client/ConnectionEstablisher.cpp

```diff
@@ -91,7 +97,15 @@ void ConnectionEstablisher::run(ConnectionEstablisher::TryResult & result, std::
 
         const UInt32 delay = table_status_it->second.absolute_delay;
         if (delay < max_allowed_delay)
+        {
             result.is_up_to_date = true;
+
+            fiu_do_on(FailPoints::replicated_merge_tree_all_replicas_stale,
+            {
+                result.delay = 1;
```

Comment:
In case anything depends on it, set the delay to a value that exceeds the max value.
```suggestion
                result.delay = max_allowed_delay + 1;
```

Reply:
I'm going to merge it now just to fix the flakiness.
In this test we don't care about the exact value of the delay, I just set it to minimal possible value to emulate some kind of ordering (which I also don't think is necessary).
Also, it's questionable if any test can rely on the exact value of the delay because it's a timestamp over which we have no complete control.